erdos/src/dataflow/operators/map.rs

```diff
@@ -95,7 +128,25 @@ where
         let op_name = format!("MapOp_{}", self.id());
 
         crate::connect_one_in_one_out(
-            move || -> MapOperator<D1, D2> { MapOperator::new(map_fn.clone()) },
+            move || -> FlatMapOperator<D1, _> {
+                let map_fn = map_fn.clone();
+                FlatMapOperator::new(move |x| std::iter::once(map_fn(x)))
+            },
+            || {},
+            OperatorConfig::new().name(&op_name),
+            self,
+        )
+    }
+
+    fn flat_map<F, I>(&self, flat_map_fn: F) -> Stream<D2>
+    where
+        F: 'static + Fn(&D1) -> I + Send + Sync + Clone,
+        I: 'static + IntoIterator<Item = D2>,
+    {
+        let op_name = format!("FlatMapOp_{}", self.name());
```

Comment:
```suggestion
        let op_name = format!("FlatMapOp_{}", self.id());
```

Reply:
Why do we prefer to use the stream ID over the stream name?

Reply:
Same problem as before, if you cascade operators via LINQ the name ends up being `MapOp_SplitOp_FilterOp_writestream_writestream_writestream`.